core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -555,6 +556,66 @@ public void testDropNamespace() {
         "is not empty. 1 tables exist.", () -> catalog.dropNamespace(tbl4.namespace()));
   }
 
+  @Test
+  public void testCreateNamespace() {
+    Namespace testNamespace = Namespace.of("testDb", "ns1", "ns2");
+    // Test with null metadata
+    AssertHelpers.assertThrows("Cannot create a namespace with null or empty metadata", IllegalArgumentException.class,
+            () -> catalog.createNamespace(testNamespace, null));
+    Assert.assertFalse(catalog.namespaceExists(testNamespace));
```

Comment:
Question for my own understanding:
In spark (let's say), when we call `CREATE DATABASE default`, what metadata gets added?

I see in the tests these two spark statements. The first seems like it could potentially have a null or empty metadata.
```
sql("CREATE NAMESPACE %s", fullNamespace);
sql("CREATE NAMESPACE %s WITH PROPERTIES ('prop'='value')", fullNamespace);
```

Are we sure that metadata can't be empty, in any engine?

Reply:
Addressing all the metadata emptiness questions: @rdblue  had a comment earlier about not allowing creation of namespaces without metadata. I think it makes sense to have this check since we are inserting metadata at the create and a record with metadata is indicative of a namespace existing.